### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Likh-Alex**
- Created model `Movie`
- Created `MovieDao` and `MovieDaoImpl`
- Created `MovieService` and `MovieServiceImpl`

### CODE REVIEW

**Review by Alex-Korzun — APPROVED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: where do we use `itemId`?

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    }
+
+    @Override
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaQuery<Movie> criteriaQuery = session.getCriteriaBuilder()
+                    .createQuery(Movie.class);
+            criteriaQuery.from(Movie.class);
+
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: remove empty line

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java (hunk 2 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: Done

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java (hunk 1 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: We do not use it. I am still not sure about setting `itemId`. As it is described in the checklist that I should not `set` id - and I followed this point, but I was not sure if this variable is generally redundant or not.
Deleting `itemId`)

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**